datalad/tests/test_auto.py

```diff
@@ -172,12 +173,43 @@ def generate_dat(f):
             f.write("123")
 
     def verify_dat(f, mode="r"):
-        with open(f, "r") as f:
+        with open(f, mode) as f:
             eq_(f.read(), "123")
 
     yield _test_proxying_open, generate_dat, verify_dat
 
 
+def test_proxying_io_open_regular():
+
+    def generate_dat(f):
+        with io.open(f, "w", encoding='utf-8') as f:
+            f.write(u"123")
```

Comment:
The `f.write(u"123")` is what is failing in PY3 build.
`write` expects sth, that implements [buffer interface](https://docs.python.org/3/c-api/buffer.html).
Since it's an ASCII string anyway `b"123"` should work in this case, I think.